File: python_backend/api_helpers/id_assign_/get_standard_ids.py
from api_helpers.id_assign_.standard_ids import _standard_ids


def get_standard_ids(exclude=[]):
    """the dict in standard_ids.py has key the name of piece, and value the id of piece.
       Reverse this definition and return dict with key the id and value the name. filter
       out names of any standard pieces being substituted by assign_ids pieces
    """
    standard_ids = {}
    names = list(filter(lambda name: name not in exclude, _standard_ids.keys()))
    for name, id_ in _standard_ids.items():
        if name in names:
            standard_ids[id_] = name
    return standard_ids

File: node.js/apiHelpers/idAssign/getStandardIds.js
const standardIds = require("./standardIds");


function getStandardIds(exclude=[]) {
    /**the standardIds.py.includes(dict) has key the name of piece, and value the id of piece.
       Reverse this definition and return dict with key the id and value the name. filter
       out names of any standard pieces being substituted by assignIds pieces
    */
    let standardIds = {}
    let names = Object.keys(standardIds).filter(name => ! exclude.includes(name))
    for (var [name, id_] of Object.entries(standardIds)) {
        if (names.includes(name)) {
            standardIds[id_] = name
        }
    }
    return standardIds
}

